content/blog/tina-cloud-and-nextjs-the-perfect-match.md

```diff
@@ -0,0 +1,108 @@
+
+```js
+const Inner = (props) => {
+// Fetch dynamic data from Tina cloud
+  const [payload, isLoading] = useGraphqlForms({
+    query: (gql) => gql(props.query),
+    variables: props.variables || {},
+  });
+  return (
+    <>
+      {isLoading ? (
+        <>
+          <LoadingPage />
+           // Return children with Static Props
+           {props.children(props)}
+        </>
+      ) : (
+        // pass the new edit state data to the child, 
+        // payload is React State and will update when the form data is updated
+        props.children({ ...props, data: payload })
+      )}
+    </>
+  );
+};
+```
+
+We needed a solution that was **statically generated** as fetching all of the data from GitHub on each request would slow down your website. We need to fetch our static data once at build time. Since the site is statically generated, the production site can be static while the editing site can load data dynamically from GitHub, directly or through our Tina Cloud Content API.
+
+For now, **we are recommending using Tina Cloud in your Next.js projects**, as it is the quickest way to get up and running. The use of `getStaticProps` allows you to fetch your static files for production builds and fetch dynamic data forms to edit your site with Tina Cloud.
+
+## Don’t let TinaCMS impact your production website
+
+Given it’s all React, it is tempting to load TinaCMS on your Next.js website and call it a day. Please bear in mind that given its client-side nature, TinaCMS comes with some JS bundles and a performance price to pay. As always it depends on your context, it might be perfectly fine for an internal project where you care much more about providing a better editing experience to your teammates than a perfect Lighthouse score. On a public-facing website though, TinaCMS should only load on your site when you're in editing mode so it does not increase your production bundle. Your visitors shouldn’t have to pay any performance fees.
+
+In edit mode, your site is wrapped with a `TinaProvider` component that uses React state to update your website in real-time. This removes the need to wait for a build step and unlocks instant previews.
+
+This is made easy with [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import#basic-usage) that will code split out the `TinaWrapper` when not in edit mode.
+
+In your Next.js `_app.js` file you can write:
+
+```js
+function InnerApp({ Component, pageProps }) {
+  const {edit} = useEditState()
+  if (edit) {
+    // Dynamically load TinaCMS when in edit mode 
+    const TinaWrapper = dynamic(() => import("../components/tina-wrapper"));
```

Comment:
Should we name the previous example `TinaWrapper` instead of `Inner` to make this a clear connection?

Reply:
I will update this